src/pages/NewsList.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useState, useEffect, ReactNode } from "react"
+import { View } from "react-native"
+
+import { api, RetryType } from "api"
+import { RootStackScreen } from "navigation/NavigationTypes"
+import { Page } from "components/Page"
+import { CardWithGradient } from "components/CardWithGradient"
+
+/**
+ * News page containing the articles of a specific category.
+ */
+export const NewsList: RootStackScreen<"NewsList"> = props => {
+    // Name of the news category, which is the title of the page
+    const { categoryName } = props.route.params
+
+    // State of the toggle switch in the header (on / off)
+    const [toggled, setToggled] = useState<boolean>(false)
+
+    const [articlesCards, setArticlesCards] = useState<ReactNode[]>([])
+
+    const [refreshing, setRefreshing] = useState<boolean>(false)
+
+    const updateArticles = (retryType: RetryType) => {
+        api.getArticles(retryType)
+            .then(articles => {
```

Comment:
usa `async/await` dove puoi, imo è più leggibile
qualcosa tipo
```suggestion
    const updateArticles = async (retryType: RetryType) => {
        const articles = await api.getArticles(retryType)
```

se la chiamata può throware wrappa in un blocco `try/catch` ma credo che di default ritenti all'infinito senza mai throware, non credo serva qua